opentelemetry-api/tests/baggage/test_baggage_propagation.py

```diff
@@ -106,6 +106,29 @@ def test_header_contains_pair_too_long(self):
         expected = {"key1": "value1", "key3": "value3"}
         self.assertEqual(self._extract(header), expected)
 
+    def test_header_max_entries_skip_invalid_entry(self):
+
+        self.assertEqual(
+            self._extract(
+                ",".join(
+                    [
+                        f"key{index}=value{index}"
+                        if index != 2
+                        else (
+                            f"key{index}="
+                            f"value{'s' * (W3CBaggagePropagator._MAX_PAIR_LENGTH + 1)}"
+                        )
+                        for index in range(W3CBaggagePropagator._MAX_PAIRS + 1)
+                    ]
+                )
+            ),
+            {
+                f"key{index}": f"value{index}"
+                for index in range(W3CBaggagePropagator._MAX_PAIRS + 1)
+                if index != 2
+            },
+        )
```

Comment:
Did I misunderstand the issue? Shouldn't the test be about the case where exception is raised but still counted? How is this test doing it?

Reply:
An exception being raised is not the only situation where a discarded entry was being counted, entries which are too long were also being discarded and being counted. This test makes sure that a discarded entry is no longer counted by introducing an entry that is too long in a sequence with one entry too many. In this way the spot of the too long entry is taken now by the entry that made the sequence have one entry too many, like this:

``` python
max_pairs = 5
max_pair_length = 11

header = ",".join(
    [
        f"key{index}=value{index}" if index != 2 else (
            f"key{index}="
            f"value{'s' * (max_pair_length + 1)}"
        ) for index in range(max_pairs + 1)
    ]
)

print(f"header: {header}")


def extract_before(header):

    context = {}
    total_baggage_entries = max_pairs

    for entry in header.split(","):
        if total_baggage_entries <= 0:
            return context

        total_baggage_entries -= 1

        if len(entry) > max_pair_length:
            continue

        name, value = entry.split("=", 1)
        context[name] = value

    return context


def extract_after(header):

    context = {}
    total_baggage_entries = max_pairs
    for entry in header.split(","):
        if len(entry) > max_pair_length:
            continue

        name, value = entry.split("=", 1)
        context[name] = value

        total_baggage_entries -= 1

        if total_baggage_entries <= 0:
            break

    return context


expected = {
    f"key{index}": f"value{index}"
    for index in range(max_pairs + 1)
    if index != 2
}

print(f"expected: {expected}")

actual = extract_after(header)
assert actual == expected, f"{actual} {expected}"

actual = extract_before(header)
assert actual == expected, f"{actual} {expected}"
```
```
header: key0=value0,key1=value1,key2=valuessssssssssss,key3=value3,key4=value4,key5=value5
expected: {'key0': 'value0', 'key1': 'value1', 'key3': 'value3', 'key4': 'value4', 'key5': 'value5'}
Traceback (most recent call last):
  File "/home/ocelotl/sandbox/test.py", line 67, in <module>
    assert actual == expected, f"{actual} {expected}"
AssertionError: {'key0': 'value0', 'key1': 'value1', 'key3': 'value3', 'key4': 'value4'} {'key0': 'value0', 'key1': 'value1', 'key3': 'value3', 'key4': 'value4', 'key5': 'value5'}
```

Reply:
Thanks, I understand now.

Reply:
I have also added a test for the case where an entry is being discarded because an exception is being raised :v: